app/models/User.coffee

```diff
@@ -5,6 +5,21 @@ ThangType = require './ThangType'
 Level = require './Level'
 utils = require 'core/utils'
 
+# Pure functions for use in Vue
+# First argument is always a raw User.attributes
+# Accessible via eg. `User.broadName(userObj)`
+UserLib = {
+  broadName: (user) ->
+    return '(deleted)' if user.deleted
+    name = _.filter([user.firstName, user.lastName]).join(' ')
+    return name if name
+    name = user.name
+    return name if name
+    [emailName, emailDomain] = user.email?.split('@') or []
+    return emailName if emailName
+    return 'Anonymous'
+}
```

Comment:
Why not do this sort of thing as just CoffeeScript class methods instead of defining it outside the class and then assigning it in?

```coffeescript
 module.exports = class User extends CocoModel
  # ...
  @broadName: (user) ->
    # ...
```